### PR TITLE
fix: Attempt to save `ParseObject` even if its nested `ParseObject` failed to save

### PR DESCRIPTION
**Pull request by Nidal-Bakir**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description

Closes: #854

### Approach

Consider the `totalResponse` from the batch as unsuccessful
``` dart 

// if any request in a batch requests group fails,
// then the overall response will be considered unsuccessful.
totalResponse.success = false;

```

### TODOs before merging

Merge #858 first, then I will add a change log entry with that PR, so we publish two fixes in one version.(4.0.1)

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry

### CODE REVIEW

**Comment by Nidal-Bakir** [comment]
The dart test will not pass because the other PR has not been merged yet, the two bugs are related.

**Comment by Nidal-Bakir** [comment]
Ready to be merged.

**Review by mbfakourii — DISMISSED** [review]

File: packages/dart/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [4.0.2](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-4.0.0...dart-4.0.1) (2023-03-21)
+
+### Bug Fixes
+
+* The SDK will try to save a ParseObject even if its child ParseObject not saved ([#859](https://github.com/parse-community/Parse-SDK-Flutter/pull/859))
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:5]:
```suggestion
* Attempt to save `ParseObject` even if its nested `ParseObject` failed to save ([#859](https://github.com/parse-community/Parse-SDK-Flutter/pull/859))
```

File: packages/dart/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## [4.0.2](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-4.0.0...dart-4.0.1) (2023-03-21)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]:
```suggestion
## [4.0.2](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-4.0.0...dart-4.0.1) (2023-03-23)
```

**Review by mtrezza — APPROVED** [review]
Looks good! Waiting for CI ...

**Pull request merged**